sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowCreateTableExec.scala

```diff
@@ -0,0 +1,120 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.v2
+
+import scala.collection.mutable
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.catalyst.util.escapeSingleQuotedString
+import org.apache.spark.sql.connector.catalog.{CatalogV2Util, Table, TableCatalog}
+import org.apache.spark.sql.execution.LeafExecNode
+import org.apache.spark.unsafe.types.UTF8String
+
+/**
+ * Physical plan node for show create table.
+ */
+case class ShowCreateTableExec(
+    output: Seq[Attribute],
+    table: Table) extends V2CommandExec with LeafExecNode {
+  override protected def run(): Seq[InternalRow] = {
+    val builder = StringBuilder.newBuilder
+    // it is used to generate Spark DDL for given table. include Hive Serde table
+    showCreateTable(table, builder)
+    Seq(InternalRow(UTF8String.fromString(builder.toString)))
+  }
+
+  private def showCreateTable(table: Table, builder: StringBuilder): Unit = {
+    builder ++= s"CREATE TABLE ${table.name()} "
+
+    showTableDataColumns(table, builder)
+    showTableUsing(table, builder)
+    showTableOptions(table, builder)
+    showTablePartitioning(table, builder)
+    showTableComment(table, builder)
+    showTableLocation(table, builder)
+    showTableProperties(table, builder)
+  }
+
+  private def showTableDataColumns(table: Table, builder: StringBuilder): Unit = {
+    val columns = table.schema().fields.map(_.toDDL)
+    builder ++= concatByMultiLines(columns)
+  }
+
+  private def showTableUsing(table: Table, builder: StringBuilder): Unit = {
+    Option(table.properties.get(TableCatalog.PROP_PROVIDER))
+      .map("USING " + escapeSingleQuotedString(_) + "\n")
+      .foreach(builder.append)
+  }
+
+  private def showTableOptions(table: Table, builder: StringBuilder): Unit = {
+    import scala.collection.JavaConverters._
+    val dataSourceOptions = table.properties.asScala
+      .filterKeys(_.startsWith(TableCatalog.OPTION_PREFIX))
+    if (dataSourceOptions.nonEmpty) {
+      val props = dataSourceOptions.map { case (key, value) =>
+        s"'${escapeSingleQuotedString(key)}' = '${escapeSingleQuotedString(value)}'"
+      }
+
+      builder ++= "OPTIONS"
+      builder ++= concatByMultiLines(props)
+    }
+  }
+
+  private def showTablePartitioning(table: Table, builder: StringBuilder): Unit = {
+    if (!table.partitioning.isEmpty) {
+      val transforms = new mutable.ArrayBuffer[String]
+      table.partitioning.foreach(t => transforms += t.describe())
+      if (transforms.nonEmpty) {
```

Comment:
this is is not needed, as it's already covered by `if (!table.partitioning.isEmpty)`

Reply:
yes